lib/binding.js

```diff
@@ -450,6 +450,9 @@ Binding.prototype.stat = function(filepath, options, callback, ctx) {
   markSyscall(ctx, 'stat');
 
   return maybeCallback(wrapStatsCallback(callback), ctx, this, function() {
+    if (Buffer.isBuffer(filepath)) {
+      filepath = filepath.toString();
+    }
```

Comment:
Suggest refactoring the Buffer stringification out to a utility function. Then, all other places would use the function's result in their code flow.
```suggestion
function stringifyFilepath (filepath) {
    if (Buffer.isBuffer(filepath)) {
      filepath = filepath.toString();
    }
    return filepath || "";
}

...
    // Somewhere else in the code
    pathname = stringifyFilepath(pathname);
```

Reply:
Thx! Refactored. I thought about it, but was lazy :)

Skipped `|| ""` because we want exception to report original user input, not `""` (which will confuse user when debugging their code).